components/datetime/src/arithmetic.rs

```diff
@@ -81,6 +81,57 @@ fn test_iso_date_to_weekday() {
     assert_eq!(IsoWeekday::Wednesday, iso_date_to_weekday(-379, 1, 2));
 }
 
+fn is_leap_year(year: i32) -> bool {
```

Comment:
thought: a lot of this math can be unified with functions that exist in the icu_calendar crate, which is already a dependency here. I'll be doing some unification work so it doesn't have to be done now.